diff --git a/modules/flowable-engine/src/main/java/org/flowable/engine/impl/bpmn/behavior/InclusiveGatewayActivityBehavior.java b/modules/flowable-engine/src/main/java/org/flowable/engine/impl/bpmn/behavior/InclusiveGatewayActivityBehavior.java
index 2e662e50f0..70779d512b 100644
--- a/modules/flowable-engine/src/main/java/org/flowable/engine/impl/bpmn/behavior/InclusiveGatewayActivityBehavior.java
+++ b/modules/flowable-engine/src/main/java/org/flowable/engine/impl/bpmn/behavior/InclusiveGatewayActivityBehavior.java
@@ -12,10 +12,14 @@
  */
 package org.flowable.engine.impl.bpmn.behavior;
 
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Iterator;
+import java.util.List;
 import java.util.Objects;
 
+import org.flowable.bpmn.model.Activity;
+import org.flowable.bpmn.model.FlowElement;
 import org.flowable.bpmn.model.FlowNode;
 import org.flowable.common.engine.impl.context.Context;
 import org.flowable.common.engine.impl.interceptor.CommandContext;
@@ -66,6 +70,11 @@ public class InclusiveGatewayActivityBehavior extends GatewayActivityBehavior im
 
         lockFirstParentScope(execution);
 
+        DelegateExecution multiInstanceExecution = null;
+        if (hasMultiInstanceParent((FlowNode) execution.getCurrentFlowElement())) {
+            multiInstanceExecution = findMultiInstanceParentExecution(execution);
+        }
+
         Collection<ExecutionEntity> allExecutions = executionEntityManager.findChildExecutionsByProcessInstanceId(execution.getProcessInstanceId());
         Iterator<ExecutionEntity> executionIterator = allExecutions.iterator();
         boolean oneExecutionCanReachGatewayInstance = false;
@@ -73,34 +82,31 @@ public class InclusiveGatewayActivityBehavior extends GatewayActivityBehavior im
             ExecutionEntity executionEntity = executionIterator.next();
             if (!executionEntity.getActivityId().equals(execution.getCurrentActivityId())) {
                 if (ExecutionGraphUtil.isReachable(execution.getProcessDefinitionId(), executionEntity.getActivityId(), execution.getCurrentActivityId())) {
-                    //Now check if they are in the same "execution path"
-                    if (executionEntity.getParentId().equals(execution.getParentId())) {
+                    if (isChildOfSameScopeExecution(executionEntity, execution, multiInstanceExecution)) {
                         oneExecutionCanReachGatewayInstance = true;
                         break;
                     }
                 }
             } else if (executionEntity.isActive() && (executionEntity.getId().equals(execution.getId()) || isAsynchronousActivity(executionEntity))) {
-                // Special case: the execution has reached the inc gw, but the operation hasn't been executed yet for that execution
-                oneExecutionCanReachGatewayInstance = true;
-                break;
+                if (isChildOfSameScopeExecution(executionEntity, execution, multiInstanceExecution)) {
+                    oneExecutionCanReachGatewayInstance = true;
+                    break;
+                }
             }
         }
 
-        // Is needed to set the endTime for all historic activity joins
         if (!inactiveCheck || !oneExecutionCanReachGatewayInstance) {
             CommandContextUtil.getActivityInstanceEntityManager(commandContext).recordActivityEnd(execution, null);
         }
 
-        // If no execution can reach the gateway, the gateway activates and executes fork behavior
         if (!oneExecutionCanReachGatewayInstance) {
 
             LOGGER.debug("Inclusive gateway cannot be reached by any execution and is activated");
 
-            // Kill all executions here (except the incoming)
             Collection<ExecutionEntity> executionsInGateway = executionEntityManager
                 .findInactiveExecutionsByActivityIdAndProcessInstanceId(execution.getCurrentActivityId(), execution.getProcessInstanceId());
             for (ExecutionEntity executionEntityInGateway : executionsInGateway) {
-                if (!executionEntityInGateway.getId().equals(execution.getId()) && executionEntityInGateway.getParentId().equals(execution.getParentId())) {
+                if (!executionEntityInGateway.getId().equals(execution.getId()) && isInSameScopeAs(executionEntityInGateway, execution, multiInstanceExecution)) {
 
                     if (!Objects.equals(executionEntityInGateway.getActivityId(), execution.getActivityId())) {
                         CommandContextUtil.getActivityInstanceEntityManager(commandContext).recordActivityEnd(executionEntityInGateway, null);
@@ -110,7 +116,6 @@ public class InclusiveGatewayActivityBehavior extends GatewayActivityBehavior im
                 }
             }
 
-            // Leave
             CommandContextUtil.getAgenda(commandContext).planTakeOutgoingSequenceFlowsOperation(execution, true);
         }
     }
@@ -118,4 +123,72 @@ public class InclusiveGatewayActivityBehavior extends GatewayActivityBehavior im
     protected boolean isAsynchronousActivity(ExecutionEntity executionEntity) {
         return executionEntity.getCurrentFlowElement() instanceof FlowNode && ((FlowNode) executionEntity.getCurrentFlowElement()).isAsynchronous();
     }
+
+    protected boolean hasMultiInstanceParent(FlowNode flowNode) {
+        boolean hasMultiInstanceParent = false;
+        if (flowNode.getSubProcess() != null) {
+            if (flowNode.getSubProcess().getLoopCharacteristics() != null) {
+                hasMultiInstanceParent = true;
+            } else {
+                boolean hasNestedMultiInstanceParent = hasMultiInstanceParent(flowNode.getSubProcess());
+                if (hasNestedMultiInstanceParent) {
+                    hasMultiInstanceParent = true;
+                }
+            }
+        }
+        return hasMultiInstanceParent;
+    }
+
+    protected DelegateExecution findMultiInstanceParentExecution(DelegateExecution execution) {
+        DelegateExecution multiInstanceExecution = null;
+        DelegateExecution parentExecution = execution.getParent();
+        if (parentExecution != null && parentExecution.getCurrentFlowElement() != null) {
+            FlowElement flowElement = parentExecution.getCurrentFlowElement();
+            if (flowElement instanceof Activity) {
+                if (((Activity) flowElement).getLoopCharacteristics() != null) {
+                    multiInstanceExecution = parentExecution;
+                }
+            }
+
+            if (multiInstanceExecution == null) {
+                DelegateExecution potentialMultiInstanceExecution = findMultiInstanceParentExecution(parentExecution);
+                if (potentialMultiInstanceExecution != null) {
+                    multiInstanceExecution = potentialMultiInstanceExecution;
+                }
+            }
+        }
+        return multiInstanceExecution;
+    }
+
+    protected boolean isChildOfSameScopeExecution(DelegateExecution executionEntity, DelegateExecution execution, DelegateExecution multiInstanceExecution) {
+        if (multiInstanceExecution != null) {
+            return isChildOfMultiInstanceExecution(executionEntity, multiInstanceExecution)
+                    && isChildOfMultiInstanceExecution(execution, multiInstanceExecution);
+        }
+        return isInSameScopeAs(executionEntity, execution, null);
+    }
+
+    protected boolean isInSameScopeAs(DelegateExecution executionEntity, DelegateExecution execution, DelegateExecution multiInstanceExecution) {
+        if (multiInstanceExecution != null) {
+            return isChildOfMultiInstanceExecution(executionEntity, multiInstanceExecution)
+                    && isChildOfMultiInstanceExecution(execution, multiInstanceExecution);
+        }
+        return executionEntity.getParentId().equals(execution.getParentId());
+    }
+
+    protected boolean isChildOfMultiInstanceExecution(DelegateExecution executionEntity, DelegateExecution multiInstanceExecution) {
+        boolean isChild = false;
+        DelegateExecution parentExecution = executionEntity.getParent();
+        if (parentExecution != null) {
+            if (parentExecution.getId().equals(multiInstanceExecution.getId())) {
+                isChild = true;
+            } else {
+                boolean isNestedChild = isChildOfMultiInstanceExecution(parentExecution, multiInstanceExecution);
+                if (isNestedChild) {
+                    isChild = true;
+                }
+            }
+        }
+        return isChild;
+    }
 }
diff --git a/modules/flowable-engine/src/test/java/org/flowable/engine/test/bpmn/gateway/InclusiveGatewayTest.java b/modules/flowable-engine/src/test/java/org/flowable/engine/test/bpmn/gateway/InclusiveGatewayTest.java
index 4cbb332879..9834de4b0d 100644
--- a/modules/flowable-engine/src/test/java/org/flowable/engine/test/bpmn/gateway/InclusiveGatewayTest.java
+++ b/modules/flowable-engine/src/test/java/org/flowable/engine/test/bpmn/gateway/InclusiveGatewayTest.java
@@ -1252,6 +1252,78 @@ public class InclusiveGatewayTest extends PluggableFlowableTestCase {
         assertProcessEnded(processInstance.getId());
     }
 
+    @Test
+    @Deployment(resources = { "org/flowable/engine/test/bpmn/gateway/InclusiveGatewayTest.parallelSubProcessesWithInclusiveGateway.bpmn20.xml" })
+    public void testParallelSubProcessesWithInclusiveGateway() {
+        ProcessInstance processInstance = runtimeService.startProcessInstanceByKey("parallelSubProcessesWithInclusiveGateway");
+
+        List<Execution> childExecutions = runtimeService.createExecutionQuery().processInstanceId(processInstance.getId()).onlyChildExecutions().list();
+        Map<String, List<Execution>> classifiedExecutions = childExecutions.stream().collect(Collectors.groupingBy(Execution::getActivityId));
+        assertThat(classifiedExecutions)
+                .containsKeys("subProcessA", "subProcessB", "nestedSubProcessA", "nestedSubProcessB",
+                        "taskA1", "taskA2", "taskB1", "taskB2");
+        assertThat(classifiedExecutions.get("subProcessA")).hasSize(1);
+        assertThat(classifiedExecutions.get("subProcessB")).hasSize(1);
+        assertThat(classifiedExecutions.get("nestedSubProcessA")).hasSize(1);
+        assertThat(classifiedExecutions.get("nestedSubProcessB")).hasSize(1);
+        assertThat(classifiedExecutions.get("taskA1")).hasSize(1);
+        assertThat(classifiedExecutions.get("taskA2")).hasSize(1);
+        assertThat(classifiedExecutions.get("taskB1")).hasSize(1);
+        assertThat(classifiedExecutions.get("taskB2")).hasSize(1);
+
+        List<Task> tasks = taskService.createTaskQuery().processInstanceId(processInstance.getId()).list();
+        Map<String, List<Task>> classifiedTasks = tasks.stream().collect(Collectors.groupingBy(Task::getTaskDefinitionKey));
+        assertThat(classifiedTasks)
+                .containsOnlyKeys("taskA1", "taskA2", "taskB1", "taskB2");
+        assertThat(classifiedTasks.get("taskA1")).hasSize(1);
+        assertThat(classifiedTasks.get("taskA2")).hasSize(1);
+        assertThat(classifiedTasks.get("taskB1")).hasSize(1);
+        assertThat(classifiedTasks.get("taskB2")).hasSize(1);
+
+        taskService.complete(classifiedTasks.get("taskA1").get(0).getId());
+
+        childExecutions = runtimeService.createExecutionQuery().processInstanceId(processInstance.getId()).onlyChildExecutions().list();
+        classifiedExecutions = childExecutions.stream().collect(Collectors.groupingBy(Execution::getActivityId));
+        assertThat(classifiedExecutions)
+                .containsKeys("subProcessA", "subProcessB", "nestedSubProcessA", "nestedSubProcessB",
+                        "taskA1", "taskA2", "taskB1", "taskB2", "inclusiveJoinA");
+        assertThat(classifiedExecutions.get("taskA1")).hasSize(0);
+        assertThat(classifiedExecutions.get("taskA2")).hasSize(1);
+        assertThat(classifiedExecutions.get("taskB1")).hasSize(1);
+        assertThat(classifiedExecutions.get("taskB2")).hasSize(1);
+        assertThat(classifiedExecutions.get("inclusiveJoinA")).hasSize(1);
+
+        taskService.complete(taskService.createTaskQuery().processInstanceId(processInstance.getId()).taskDefinitionKey("taskA2").singleResult().getId());
+        taskService.complete(taskService.createTaskQuery().processInstanceId(processInstance.getId()).taskDefinitionKey("taskB1").singleResult().getId());
+        taskService.complete(taskService.createTaskQuery().processInstanceId(processInstance.getId()).taskDefinitionKey("taskB2").singleResult().getId());
+
+        childExecutions = runtimeService.createExecutionQuery().processInstanceId(processInstance.getId()).onlyChildExecutions().list();
+        classifiedExecutions = childExecutions.stream().collect(Collectors.groupingBy(Execution::getActivityId));
+        assertThat(classifiedExecutions)
+                .containsKeys("subProcessA", "subProcessB", "nestedSubProcessA", "nestedSubProcessB",
+                        "postForkTaskA", "postForkTaskB");
+        assertThat(classifiedExecutions.get("postForkTaskA")).hasSize(1);
+        assertThat(classifiedExecutions.get("postForkTaskB")).hasSize(1);
+
+        tasks = taskService.createTaskQuery().processInstanceId(processInstance.getId()).list();
+        classifiedTasks = tasks.stream().collect(Collectors.groupingBy(Task::getTaskDefinitionKey));
+        assertThat(classifiedTasks)
+                .containsOnlyKeys("postForkTaskA", "postForkTaskB");
+        assertThat(classifiedTasks.get("postForkTaskA")).hasSize(1);
+        assertThat(classifiedTasks.get("postForkTaskB")).hasSize(1);
+
+        tasks.forEach(this::completeTask);
+
+        childExecutions = runtimeService.createExecutionQuery().processInstanceId(processInstance.getId()).onlyChildExecutions().list();
+        assertThat(childExecutions).hasSize(1);
+        Task task = taskService.createTaskQuery().processInstanceId(processInstance.getId()).singleResult();
+        assertThat(task.getTaskDefinitionKey()).isEqualTo("lastTask");
+
+        taskService.complete(task.getId());
+
+        assertThat(runtimeService.createProcessInstanceQuery().processInstanceId(processInstance.getId()).count()).isZero();
+    }
+
     @Test
     @Deployment
     void testWithFutureDelegates() {
