diff --git a/modules/flowable-process-validation/src/main/java/org/flowable/validation/validator/Problems.java b/modules/flowable-process-validation/src/main/java/org/flowable/validation/validator/Problems.java
index 1da8cb6dd0..ea244cd0f8 100644
--- a/modules/flowable-process-validation/src/main/java/org/flowable/validation/validator/Problems.java
+++ b/modules/flowable-process-validation/src/main/java/org/flowable/validation/validator/Problems.java
@@ -139,5 +139,11 @@ public interface Problems {
     String DI_INVALID_REFERENCE = "flowable-di-invalid-reference";
     String DI_DOES_NOT_REFERENCE_FLOWNODE = "flowable-di-does-not-reference-flownode";
     String DI_DOES_NOT_REFERENCE_SEQ_FLOW = "flowable-di-does-not-reference-seq-flow";
+    
+    String MISSING_ELEMENT_VARIABLE = "flowable-multi-instance-missing-element-variable";
+    String ASSIGNEE_NOT_USING_ELEMENT_VARIABLE = "flowable-assignee-not-using-element-variable";
+    String CANDIDATE_USERS_NOT_USING_ELEMENT_VARIABLE = "flowable-candidate-users-not-using-element-variable";
+    String CANDIDATE_GROUPS_NOT_USING_ELEMENT_VARIABLE = "flowable-candidate-groups-not-using-element-variable";
+    String COLLECTION_AND_LOOPCARDINALITY_CONFLICT = "flowable-collection-and-loopcardinality-conflict";
 
 }
diff --git a/modules/flowable-process-validation/src/main/java/org/flowable/validation/validator/ValidatorSetFactory.java b/modules/flowable-process-validation/src/main/java/org/flowable/validation/validator/ValidatorSetFactory.java
index 544836c3c3..973ddd0d9a 100644
--- a/modules/flowable-process-validation/src/main/java/org/flowable/validation/validator/ValidatorSetFactory.java
+++ b/modules/flowable-process-validation/src/main/java/org/flowable/validation/validator/ValidatorSetFactory.java
@@ -37,6 +37,7 @@ import org.flowable.validation.validator.impl.ServiceTaskValidator;
 import org.flowable.validation.validator.impl.SignalValidator;
 import org.flowable.validation.validator.impl.StartEventValidator;
 import org.flowable.validation.validator.impl.SubprocessValidator;
+import org.flowable.validation.validator.impl.MultiInstanceUserTaskValidator;
 import org.flowable.validation.validator.impl.UserTaskValidator;
 
 /**
@@ -59,6 +60,7 @@ public class ValidatorSetFactory {
         validatorSet.addValidator(new StartEventValidator());
         validatorSet.addValidator(new SequenceflowValidator());
         validatorSet.addValidator(new UserTaskValidator());
+        validatorSet.addValidator(new MultiInstanceUserTaskValidator());
         validatorSet.addValidator(new ServiceTaskValidator());
         validatorSet.addValidator(new ScriptTaskValidator());
         validatorSet.addValidator(new SendTaskValidator());
